app/helpers/proxmox_vm_cloudinit_helper.rb

```diff
@@ -41,4 +41,81 @@ def parse_server_cloudinit(args)
     end
     cloudinit_h
   end
+
+  def create_cloudinit_iso(vm_name, configs)
+    ssh = Fog::SSH.new(URI.parse(fog_credentials[:proxmox_url]).host, fog_credentials[:proxmox_username].split('@')[0], { password: fog_credentials[:proxmox_password] })
+    files = []
+    wd = create_temp_directory(ssh)
+    configs.each do |config|
+      begin
+        ssh.run([%({echo "#{config[1]}" >> "#{wd}/#{config[0]}"})])
+      rescue ::Foreman::Exception => e
+        logger.error("Error writing to the file #{config[0]}: #{e}")
+      end
+      files.append(File.join(wd, config[0]))
+    end
+    raise Foreman::Exception, N_('ISO build failed') unless ssh.run(generate_iso_command(vm_name, files))
+    delete_config_files(wd)
+    iso
+  end
+
+  def generate_iso_command(vm_name, files)
+    iso = File.join(default_iso_path, "#{vm_name}_cloudinit.iso")
+    arguments = ["genisoimage", "-output #{iso}", '-volid', 'cidata', '-joliet', '-rock']
+    logger.warn("iso image generation args: #{arguments}")
+    arguments.concat(files).join(' ')
+  end
+
+  def create_temp_directory(ssh)
+    res = ssh.run([%({mktemp -d})])
+    if res.first.status.zero?
+      wd = res.first.stdout.chomp
+    else
+      logger.warn("Could not create working directory to store cloudinit config data: #{res.first.stdout}. Creating files directly in '/tmp' directory.")
```

Comment:
if we can not create the temp directory, I would suggest to raise an exception and fail completly.